urllib3/exceptions.py

```diff
@@ -180,6 +180,14 @@ class SNIMissingWarning(HTTPWarning):
     pass
 
 
+class DependencyWarning(HTTPWarning):
```

Comment:
I know `HTTPWarning` is our base warning, but it is irksome here. Not sure we can fix this pre-2.0 though.